consensus/reactor.go

```diff
@@ -196,6 +196,18 @@ func (r *Reactor) OnStart() error {
 // blocking until they all exit, as well as unsubscribing from events and stopping
 // state.
 func (r *Reactor) OnStop() {
+
+	// If the node is committing the new block, wait until it finished!
+	var c = 0
+	for (r.state.Step == cstypes.RoundStepCommit) && (c < 3) {
+		time.Sleep(time.Second)
+		c++
+	}
+
+	if c == 3 {
+		r.Logger.Error("the 3 secs block commit waiting timeout! The block commit might be failed!")
+	}
+
```

Comment:
The idea behind this is that **only** if we are in the `CommitStep` do we wait for that step to finish, not any other step right? 

Could we not add a listener for a `RoundStepNewHeight` event and wait for that to fire before continuing

Reply:
Right, when the consensus is entering the commit stage, the Step will be updated
https://github.com/tendermint/tendermint/blob/05bc2f344a20bd25beb4b6f9e6f41d2295d99492/consensus/state.go#L1487
and if commit block success, the Step will be updated again
https://github.com/tendermint/tendermint/blob/05bc2f344a20bd25beb4b6f9e6f41d2295d99492/consensus/state.go#L1676

Reply:
Thanks for the suggestion, I will check it.

Reply:
Yeah would prefer to see if we can wait for an event or something to fire first.